File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.java
import java.util.Arrays;

class Solution {
    public int maxFrequency(int[] nums, int k) {
        Arrays.sort(nums); 
        int left = 0;
        int right = 0;
        int n = nums.length;
        long sum = 0; 

        int maxFrequency = 0;

        while (right < n) {
            sum += nums[right];

            long requiredSum = (long) nums[right] * (right - left + 1);
            long diff = requiredSum - sum;

            while (diff > k) {
                sum -= nums[left];
                left++;
                requiredSum = (long) nums[right] * (right - left + 1);
                diff = requiredSum - sum;
            }

            maxFrequency = Math.max(maxFrequency, right - left + 1);

            right++;
        }

        return maxFrequency;
    }
}

 

File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.cpp
class Solution {
public:
    int maxFrequency(std::vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int left = 0, right = 0;
        int n = nums.size();
        long long sum = 0;  
        int maxFrequency = 0; 

        while (right < n) {
            sum += nums[right];

            long long requiredSum = static_cast<long long>(nums[right]) * (right - left + 1);
            long long diff = requiredSum - sum;

            while (diff > k) {
                sum -= nums[left];
                left++;
                requiredSum = static_cast<long long>(nums[right]) * (right - left + 1);
                diff = requiredSum - sum;
            }
            maxFrequency = max(maxFrequency, right - left + 1);

            right++;
        }

        return maxFrequency;
    }
};